client/init.c

```diff
@@ -205,8 +205,11 @@ TDNFRefreshSack(
         pTdnf->pArgs->nRefresh = 1;
     }
 
+    dwError = TDNFInitCmdLineRepo(pTdnf, pSack);
+    BAIL_ON_TDNF_ERROR(dwError);
+
     //If there is an empty repo directory, do nothing
-    if(pTdnf->pRepos)
+    if(pTdnf->pRepos->pNext)
```

Comment:
List with single a node will not enter this case now. Is this indented?

Reply:
Yes, this is intentional. The first repo will be the "@cmdline" repo, which always exists and is the first in the list. It does not need to processed in the loop.

But when I looked over the changes, I noticed that the "@cmdline" repo is still processed in the loop, so I just changed it such that the loop starts with the second (first "regular") repo.